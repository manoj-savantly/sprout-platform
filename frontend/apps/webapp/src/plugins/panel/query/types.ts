import { QueryParametersConfiguration } from './editors/QueryParametersEditor';
import { CodeTemplateEditorConfiguration } from './editors/TemplateEditor';

export type QueryPanelDisplayType = 'FRAME' | 'RENDER';

export interface QueryPanelOptions {
  url: string;
  queryParameters: QueryParametersConfiguration;
  displayType: QueryPanelDisplayType;
  template: CodeTemplateEditorConfiguration;
  useTemplate: boolean;
  useProxy: boolean;
  headers: string;
}
