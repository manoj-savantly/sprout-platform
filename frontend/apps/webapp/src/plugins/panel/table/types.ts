import { QueryParametersConfiguration } from './editors/QueryParametersEditor';
import { CodeTemplateEditorConfiguration } from './editors/TemplateEditor';

export interface TablePanelOptions {
  url: string;
  queryParameters: QueryParametersConfiguration;
  jsonDataPath: string;
  headerTemplate: CodeTemplateEditorConfiguration;
  useProxy: boolean;
  headers: string;
}
