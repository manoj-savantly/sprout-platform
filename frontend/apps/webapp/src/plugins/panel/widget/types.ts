export interface WidgetPanelOptions {
  dataSourceId: string
  dataId: string
}

export type WidgetDataType = 'MARKDOWN' | 'MARKUP' | 'JSON'

export interface WidgetData {
    id: string;
    name: string;
    dataType: WidgetDataType;
    data: any;
}
