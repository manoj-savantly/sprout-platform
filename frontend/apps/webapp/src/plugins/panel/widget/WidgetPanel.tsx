// Libraries
import { PanelProps } from '@savantly/sprout-api';
import { CustomScrollbar, stylesFactory } from '@savantly/sprout-ui';
import { LoadingIcon, MarkdownViewer } from '@sprout-platform/ui';
import { css, cx } from 'emotion';
import React from 'react';
import { useWidgetDataById } from './api/widgetApi';
import { WidgetPanelOptions } from './types';

interface Props extends PanelProps<WidgetPanelOptions> {}

// TODO: change rendering based on data type
const RenderData = ({ data, dataType }: { data: any; dataType: string }) => {
  if (data) {
    return <MarkdownViewer className={cx(getStyles().content)}>{data}</MarkdownViewer>;
  } else {
    return <LoadingIcon />;
  }
};

export const WidgetPanel = (props: Props) => {
  const { dataId, dataSourceId } = props.options;
  const widgetData = useWidgetDataById(dataSourceId, dataId);

  return (
    <CustomScrollbar autoHeightMin="100%">
      {widgetData && <RenderData data={widgetData.data} dataType={widgetData.dataType} />}
    </CustomScrollbar>
  );
};

const getStyles = stylesFactory(() => {
  return {
    content: css`
      height: 100%;
      border: none;
    `
  };
});
