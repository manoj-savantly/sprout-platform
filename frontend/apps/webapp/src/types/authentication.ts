import { User } from '@savantly/sprout-api';

export interface AuthenticationState {
  loading: boolean;
  isAuthenticated: boolean;
  loginError: boolean; // Errors returned from server side
  errorMessage?: string; // Errors returned from server side
  sessionHasBeenFetched: boolean;
  sessionFetchFailed: boolean;
  logoutUrl?: string;
  user: User;
  showLogin: boolean;
}

export interface AuthenticationUpdate {
  errorMessage?: string;
  accessToken?: string;
}

export interface OAuthClientConfig {
  name: string;
  autoLogin?: boolean;
  displayName: string;
  issuerUri: string;
  clientId: string;
  scope: string;
  authorizationUrl: string;
  redirectUrl: string;
  tokenUrl: string;
  userInfoUrl: string;
}