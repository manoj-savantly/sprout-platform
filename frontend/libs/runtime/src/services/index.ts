export * from './apiSvc';
export * from './fileSvc';
export * from './locationSvc';
export * from './panelRegistrationService';
export * from './userContextSvc';