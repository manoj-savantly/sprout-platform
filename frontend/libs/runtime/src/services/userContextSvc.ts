import { UserContext } from '@savantly/sprout-api';
export interface UserContextService {
    getUserContext: () => UserContext;
}

let singletonInstance: UserContextService;

/**
 * Used during startup by Sprout to set the service so it is available
 * via the {@link getUserContextService} to the rest of the application.
 *
 * @internal
 */
export function setUserContextService(instance: UserContextService) {
  singletonInstance = instance;
}

/**
 * Used to retrieve the {@link UserContextService}
 *
 * @public
 */
export function getUserContextService(): UserContextService {
  return singletonInstance;
}
