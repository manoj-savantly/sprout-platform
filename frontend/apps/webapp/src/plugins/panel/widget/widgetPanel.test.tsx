import { PanelProps } from '@savantly/sprout-api';
import { cleanup, render, waitForElement } from '@testing-library/react';
import moxios from 'moxios';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { sproutApiSvc } from '../../../core/services/sproutApiSvc';
import { WidgetPanelOptions } from './types';
import { WidgetPanel } from './WidgetPanel';

const mockHtmlResponse = {
  url: '/api/widget-data/111/222',
  payload: {
    status: 200,
    headers: {
      'Content-Type': 'text/html',
      'widget-id': '111',
      'widget-name': 'hello-world',
      'widget-data-type': 'MARKUP'
    },
    responseText: `<h1>Hello World</h1>`
  }
};
const mockMarkupResponse = {
  url: '/api/widget-data/222/333',
  payload: {
    status: 200,
    headers: {
      'Content-Type': 'text/html',
      'widget-id': '333',
      'widget-name': 'hello-world',
      'widget-data-type': 'MARKUP'
    },
    responseText: `# Hello World`
  }
};

const getMockWidgetPanelProps = (options: WidgetPanelOptions): PanelProps<WidgetPanelOptions> => {
  return {
    data: 123 as any,
    id: 1,
    onOptionsChange: (o) => {},
    options: options,
    renderCounter: 1,
    title: 'widget',
    transparent: false,
    height: 8,
    width: 12
  };
};

beforeAll(() => {
  moxios.install(sproutApiSvc);
});

afterAll(() => {
  moxios.uninstall(sproutApiSvc);
});

afterEach(() => {
  cleanup();
});

test('renders Markup WidgetData', async () => {
  // Match against a exact URL values
  moxios.stubRequest(mockHtmlResponse.url, mockHtmlResponse.payload);
  const component = render(
    <BrowserRouter>
      <WidgetPanel
        {...getMockWidgetPanelProps({
          dataSourceId: '111',
          dataId: '222'
        })}
      />
    </BrowserRouter>
  );
  await waitForElement(() => component.getByText('Hello World'));
  expect(component.container).toMatchSnapshot();
});

test('renders Markdown WidgetData', async () => {
  // Match against a exact URL values
  moxios.stubRequest(mockMarkupResponse.url, mockMarkupResponse.payload);

  const component = render(
    <BrowserRouter>
      <WidgetPanel
        {...getMockWidgetPanelProps({
          dataSourceId: '222',
          dataId: '333'
        })}
      />
    </BrowserRouter>
  );
  await waitForElement(() => component.getByText('# Hello World'));
  expect(component.container).toMatchSnapshot();
});
