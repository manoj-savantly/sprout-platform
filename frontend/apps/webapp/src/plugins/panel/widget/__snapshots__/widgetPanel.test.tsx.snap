// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Markdown WidgetData 1`] = `
<div>
  <div
    class="custom-scrollbar"
    style="position: relative; overflow: hidden; width: 100%; height: auto; min-height: 100%; max-height: 100%;"
  >
    <div
      class="view"
      style="position: relative; overflow: scroll; margin-right: 0px; margin-bottom: 0px; min-height: calc(100% + 0px); max-height: calc(100% + 0px);"
    >
      <div>
        # Hello World
      </div>
    </div>
    <div
      class="track-horizontal"
      style="position: absolute; height: 6px; display: none;"
    >
      <div
        class="thumb-horizontal"
        style="position: relative; display: block; height: 100%;"
      />
    </div>
    <div
      class="track-vertical"
      style="position: absolute; width: 6px; display: none;"
    >
      <div
        class="thumb-vertical"
        style="position: relative; display: block; width: 100%;"
      />
    </div>
  </div>
</div>
`;

exports[`renders Markup WidgetData 1`] = `
<div>
  <div
    class="custom-scrollbar"
    style="position: relative; overflow: hidden; width: 100%; height: auto; min-height: 100%; max-height: 100%;"
  >
    <div
      class="view"
      style="position: relative; overflow: scroll; margin-right: 0px; margin-bottom: 0px; min-height: calc(100% + 0px); max-height: calc(100% + 0px);"
    >
      <div>
        <h1>
          Hello World
        </h1>
      </div>
    </div>
    <div
      class="track-horizontal"
      style="position: absolute; height: 6px; display: none;"
    >
      <div
        class="thumb-horizontal"
        style="position: relative; display: block; height: 100%;"
      />
    </div>
    <div
      class="track-vertical"
      style="position: absolute; width: 6px; display: none;"
    >
      <div
        class="thumb-vertical"
        style="position: relative; display: block; width: 100%;"
      />
    </div>
  </div>
</div>
`;
